connect_ext_ppr/utils.py

```diff
@@ -660,3 +661,47 @@ def process_ppr_file_for_delelegate_l2(sheet_name, ws):
         ws['Published'] = 'TRUE'
     elif sheet_name == 'ServicePlans':
         ws['Published'] = 'FALSE'
+
+
+def process_service_plans_for_apply_ppr_and_delegate_to_marketplaces(
+    config_json,
+    dr_marketplaces,
+    ws,
+    marketplaces_to_not_update,
+):
+    marketplace_mapping = config_json.get('marketplace_mapping', {})
+    cbc_marketplace_ids_dict = {}
+    for dr_marketplace in dr_marketplaces:
+        if cbc_marketplace_id := marketplace_mapping.get(dr_marketplace.marketplace):
+            cbc_marketplace_ids_dict[cbc_marketplace_id] = dr_marketplace.marketplace
+        else:
+            marketplaces_to_not_update.add(dr_marketplace.marketplace)
+
+    for col in ws.columns.tolist():
+        if re.match(r'^OpUnit_[A-Z]{2}$', col):
+            cbc_marketplace_id = col.split('_')[1]
+            if cbc_marketplace_id in cbc_marketplace_ids_dict:
+                cbc_marketplace_ids_dict.pop(cbc_marketplace_id)
+            else:
+                ws[col] = 'FALSE'
+    # This value updates here, because it is not convenient to return it
+    marketplaces_to_not_update.update(cbc_marketplace_ids_dict.values())
```

Comment:
I suppose, it is a bad pattern to return value this way, but didn't come up with how to do it better. Any ideas?